yb-voyager/cmd/analyzeSchema.go

```diff
@@ -921,9 +921,14 @@ sqlParsingLoop:
 			} else if matches := dollarQuoteRegex.FindStringSubmatch(currLine); matches != nil {
 				dollarQuoteFlag = 1 //denotes start of the code/body part
 				codeBlockDelimiter = matches[0]
+			} else if matches := sqlBodyBeginRegex.FindStringSubmatch(currLine); matches != nil {
+				dollarQuoteFlag = 1        //denotes start of the sql body part https://www.postgresql.org/docs/15/sql-createfunction.html#:~:text=a%20new%20session.-,sql_body,-The%20body%20of
+				codeBlockDelimiter = "END" //SQL body to determine the end of BEGIN ATOMIC ... END; sql body
 			}
 		case CODE_BLOCK_STARTED:
-			if strings.Contains(currLine, codeBlockDelimiter) {
+			if strings.Contains(currLine, codeBlockDelimiter) ||
+				strings.Contains(currLine, strings.ToLower(codeBlockDelimiter)) {
```

Comment:
Concern here is that we don't know the variety of cases possible with plpgsql, since it can almost have anything inside the body. Here the condition is getting changed to match with either same or lowercase variant of it.

Should we just handle the sqlbody func case separately(by that i mean having different if conditions here for that) and not update the existing ones.
Just want to avoid any unknowns

Reply:
> Here the condition is getting changed to match with either same or lowercase variant of it.

the same or lowercase variant is only of the `codeBlockDelimiter` variable which could be dollarQuoteRegex    = regexp.MustCompile(`(\$.*\$)`) match or the END string for sqlBody. 

But okay, I can do that in separate if condition for just END case

Reply:
done